### ISSUE TITLE
Jest 23.1 has regex issue on Windows

### ISSUE THREAD

**Issue by benhamlin** [jaredpalmer/razzle#669]
Regretfully I missed this somehow in testing the Jest 23.1 bump, but Jest 23 has issues with 
a specific regex that's part of CRA: https://github.com/facebook/jest/issues/6385. 

Razzle has the same regex: https://github.com/jaredpalmer/razzle/blob/9a58a71235357b50e230061ddb55fe2f27084593/packages/razzle/config/createJestConfig.js#L32

Seems like there are 3 options here:
1. Roll back Jest to 22.4, which has minor, but tolerable issues.
2. Make a change in Razzle to update that regex to something Jest doesn't barf on
3. Wait for Jest to fix their transform.

**Comment by benhamlin** [comment]
More fun in windows path normalization land!

The good news is that the regex issue was fixed, and no longer causes jest to puke.

The bad news is that it seems there is another issue in 23.3 (https://github.com/facebook/jest/issues/6576) where the testMatch paths provided by razzle, i.e. `<rootDir>/src/**/?(*.)(spec|test).{js,jsx,mjs}`, is finding no tests. This repros even on a brand new razzle app. It appears to be related to `<rootDir>`, as replacing that with ** works fine.

Is there a reason default razzle testMatch includes `<rootDir>/src` ? 

In the meantime this one is easy to work around by overriding testMatch in package.json.

**Comment by jaredpalmer** [comment]
Compat with CRA

**Comment by katiawheeler** [comment]
I am having this issue as well with the default regex not finding tests.

**Comment by benhamlin** [comment]
The exact workaround until Jest fixes it (and from all appearances the fix is coming soon), is to add this to your package.json:
```
 "jest": {
    "testMatch": [
      "**/__tests__/**/*.{js,jsx,mjs}",
      "**/?(*.)(spec|test).{js,jsx,mjs}"
    ]
  }
```

Works On My Box™

**Comment by benhamlin** [comment]
Jest rolled back their version of micromatch, issue is fixed in Jest 23.4. Confirmed that fresh razzle app is finding tests again without any overriding.
